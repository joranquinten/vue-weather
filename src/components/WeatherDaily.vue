<template>
  <ol>
    <li v-for="day in daily" v-bind:key="day.time">
      <h4>
        {{ day.time | weekDayFromTime | capitalizeFirstLetter }}
        {{ day.time | monthDate }}
      </h4>
      <p>{{ day.summary }}</p>
      <p class="temperature-range">
        {{
          `${day.temperatureLow} &deg;C &ndash; ${day.temperatureHigh} &deg;C`
        }}
      </p>
    </li>
  </ol>
</template>

<script>
import { format } from 'date-fns'
import nl from 'date-fns/locale/nl'

export default {
  name: 'WeatherDaily',
  props: {
    daily: {
      type: Array,
      default: () => []
    }
  },
  filters: {
    capitalizeFirstLetter: function(value) {
      value = value.toString()
      return value.charAt(0).toUpperCase() + value.slice(1)
    },
    weekDayFromTime(value) {
      return format(new Date(value), 'dddd', { locale: nl })
    },
    monthDate(value) {
      return format(new Date(value), 'D MMMM', { locale: nl })
    }
  }
}
</script>

<style scoped></style>
