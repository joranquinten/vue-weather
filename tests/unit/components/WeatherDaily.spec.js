import { shallowMount } from '@vue/test-utils'
import WeatherDaily from '../../../src/components/WeatherDaily'

const dailyMock = [
  {
    summary: 'Dag 1',
    time: 1555365600000,
    temperatureHigh: 20.41,
    temperatureLow: 8.57,
    icon: 'icon-day-one'
  },
  {
    summary: 'Dag 2',
    time: 1555452000000,
    temperatureHigh: 10.17,
    temperatureLow: -1.24,
    icon: 'icon-day-two'
  }
]

describe('WeatherDaily.vue', () => {
  it('renders without crashing', () => {
    const wrapper = shallowMount(WeatherDaily)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders props.daily when passed', () => {
    const wrapper = shallowMount(WeatherDaily, {
      propsData: {
        daily: [dailyMock]
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})
