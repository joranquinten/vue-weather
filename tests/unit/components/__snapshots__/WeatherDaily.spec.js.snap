// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDaily.vue renders props.daily when passed 1`] = `
<ol>
  <li>
    <h4>
      Invalid Date
      Invalid Date
    </h4>
    <p></p>
    <p class="temperature-range">
      undefined °C – undefined °C
    </p>
  </li>
</ol>
`;

exports[`WeatherDaily.vue renders without crashing 1`] = `<ol></ol>`;
